libs/coin-modules/coin-tezos/src/api/index.ts

```diff
@@ -38,36 +38,58 @@ function isTezosTransactionType(type: string): type is "send" | "delegate" | "un
   return ["send", "delegate", "undelegate"].includes(type);
 }
 
+async function balance(address: string): Promise<Balance<TezosAsset>[]> {
+  const value = await getBalance(address);
+  return [
+    {
+      value,
+      asset: { type: "native" },
+    },
+  ];
+}
+
 async function craft(
-  transactionIntent: TransactionIntent<TezosAsset>,
+  transactionIntent: TezosTransactionIntent,
   customFees?: bigint,
 ): Promise<string> {
   if (!isTezosTransactionType(transactionIntent.type)) {
     throw new IncorrectTypeError(transactionIntent.type);
   }
-  const fee = customFees !== undefined ? customFees : await estimate(transactionIntent);
+  const fee =
+    customFees !== undefined
+      ? { fees: customFees.toString() }
+      : await estimate(transactionIntent).then(fees => ({
+          fees: fees.value.toString(),
+          gasLimit: fees.parameters?.gasLimit?.toString() ?? undefined,
+          storageLimit: fees.parameters?.storageLimit?.toString() ?? undefined,
```

Comment:
Unnecessary `?? undefined`
```suggestion
          gasLimit: fees.parameters?.gasLimit?.toString(),
          storageLimit: fees.parameters?.storageLimit?.toString(),
```